buildsystem/codecompliance/textfiles.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2015-2015 the openage authors. See copying.md for legal info.
+
+"""
+Checks some general whitespace rules and the encoding for text files.
+"""
+
+from .util import findfiles, readfile, has_ext, BADUTF8FILES
+
+
+def find_issues(dirnames, exts):
+    """
+    checks all files ending in exts in dirnames.
+    """
+    for filename in findfiles(dirnames, exts):
+        data = readfile(filename)
+
+        if ' \n' in data or '\t\n' in data:
+            if filename.startswith('openage/') and filename.endswith('.cpp'):
+                # allow this particular error for Cython-generated files.
+                pass
+            elif filename.endswith('.gen.h') or filename.endswith('.gen.cpp'):
+                # TODO all this for now, until someone fixes the codegen.
+                pass
+            else:
+                yield "Trailing whitespace", filename
```

Comment:
maybe we can also print the linenumber, but only calculate it when the failure was found. at least add a TODO for this.